src/test/java/net/snowflake/client/jdbc/PreparedStatement1IT.java

```diff
@@ -94,6 +81,56 @@ public void testGetParameterMetaData() throws SQLException {
     }
   }
 
+  /**
+   * Test to ensure it's possible to upload Time values via stage array binding and get proper
+   * values back (SNOW-194437)
+   *
+   * <p>Ignored on GitHub Action because CLIENT_STAGE_ARRAY_BINDING_THRESHOLD parameter is not
+   * available to customers so cannot be set when running on Github Action
+   *
+   * @throws SQLException
+   */
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testInsertStageArrayBindWithTime() throws SQLException {
+    try (Connection connection = init()) {
+      Statement statement = connection.createStatement();
+      statement.execute("alter session set CLIENT_STAGE_ARRAY_BINDING_THRESHOLD=2");
+      statement.execute("create or replace table testStageBindTime (c1 time, c2 time)");
+      PreparedStatement prepSt =
+          connection.prepareStatement("insert into testStageBindTime values (?, ?)");
+      Time[][] timeValues = {
+        {new Time(0), new Time(1)},
+        {new Time(1000), new Time(Integer.MAX_VALUE)},
+        {new Time(123456), new Time(55555)}
+      };
+      for (int i = 0; i < timeValues.length; i++) {
+        prepSt.setTime(1, timeValues[i][0]);
+        prepSt.setTime(2, timeValues[i][1]);
+        prepSt.addBatch();
+      }
+      prepSt.executeBatch();
+      // check results
+      ResultSet rs = statement.executeQuery("select * from testStageBindTime");
+      for (int i = 0; i < timeValues.length; i++) {
+        rs.next();
+        assertEquals(timeValues[i][0].toString(), rs.getTime(1).toString());
```

Comment:
Times have to be converted to strings to be compared because the dates that are returned are mismatched. For instance, new Time(Integer.MAX_VALUE)) returns a time with a date on 1970-01-25, whereas the getTime() result returned has the same time on the earlier date of 1970-01-01.